modules/exoplanets/datastructure.cpp

```diff
@@ -0,0 +1,332 @@
+            double semiMajorAxis;
+
+            // The uncertainty of the semi-major axis, given as an asymmetric uncertainty
+            // with two values: the lower and upper uncertainty for the orbit size. The
+            // values are given as values relative to the size of the SemiMajorAxis. For
+            // example, the lower (first) value is computed as the lower uncertainty
+            // value divided by the value for the semi-major axis.
+            //
+            // If no value is given, there is no uncertainty and the semi-major axis value
+            // is exact.
+            std::optional<glm::dvec2> semiMajorAxisUncertainty;
+
+            // The eccentricity of the planet's orbit. If it does not exist in the data,
+            // it is given a default value of 0
+            double eccentricity;
+
+            // The inclination of the planet's orbit, given as an angle in degrees in the
+            // range 0-360. If it does not exist in the data, it is given a default value
+            // of 90 degrees
+            double inlination;
+
+            // The longitude of ascending node, that is, the angle for the direction of
+            // the point where the orbit passes through the plane of reference. Also
+            // known as the right ascension of the ascending node. The angle is given as
+            // degrees in the range 0-360. If it does not exist in the data, it is given
+            // a default value of 180 degrees
+            double ascendingNode;
+
+            // The argument of periapsis of the orbit, given as an angle in degrees in
+            // the range 0-360. It is the angle between the planet's periapsis (the point
+            // where it is closest to its star) and its ascending node, that is, where it
+            // passes through its reference plane. If it does not exist in the data, it
+            // is given a default value of 90 degrees
+            double argumentOfPeriapsis;
+
+            // The epoch to use when computing the planet's initial position in its orbit,
+            // given as an ISO 8601-like date string of the format YYYY-MM-DDTHH:MN:SS.
+            //
+            // Computed based on the \"Time of Conjunction (Transit Midpoint)\" value in
+            // the data from the exoplanet archive
+            std::string epoch;
+
+            // Time the planet takes to make a complete orbit around the host star or
+            // system, in days
+            double period;
+
+            // A rotation matrix that represents the rotation of the plane of the orbit.
+            // It is computed based on the inclination, ascending node, and argument of
+            // periapsis values. Hence, it may be derived from default values.
+            glm::dmat3 orbitPlaneRotationMatrix;
+
+            // True if default values have been used for any of the orbit parameters, due
+            // to missing data
+            bool hasUsedDefaultValues;
```

Comment:
No, rather, the opposite. It says if any of the non-optional values have been set based on default values

Most data points do not have data for e.g. the orbit inclination, argument of periapsis, or ascending node, but we always set that value to a valid one (because otherwise we would not be able to show the orbit of almost any planets).  This parameter is true if any of those variables have been set to a default value.